### ISSUE TITLE
Add generic Record<K, V> helper

### ISSUE THREAD

**Issue by abdala** [fgnass/type-assurance#1]
Is this something you would like to have here?

```typescript
const record =
  <K extends Schema, V extends Schema>(key: K, value: V) =>
  (v: unknown): v is Record<TypeFromSchema<K>, TypeFromSchema<V>> =>
    typeof v === 'object' && v !== null 
    && Object.values(v).every((y) => is(y, value)) 
    && Object.keys(v).every((y) => is(y, key))
```
Usage:

```typescript
is(obj, record(String, String))
is(obj, record(String, union(String, Number)))
etc...
```

Nice lib! Thank you for sharing :)

**Comment by fgnass** [comment]
That sounds indeed useful, I wonder why did not think of this before!

**Comment by abdala** [comment]
OK. I'll open a PR then, ok?

**Comment by fgnass** [comment]
Looking forward to it!

**Comment by fgnass** [comment]
Thank you! I published v1.6.0 containing this feature.
